File: C/1480.c
/**
 * Note: The returned array must be malloced, assume caller calls free().
 */
int *runningSum(int *nums, int numsSize, int *returnSize)
{

    // Validate the given input.
    if (nums == NULL || numsSize == 0)
    {
        printf("Invalid input.\n");
        return NULL;
    }

    *returnSize = numsSize;

    int *answer = (int *)malloc(numsSize * sizeof(int));

    // Validate if malloc succeded.
    if (answer == NULL)
    {
        printf("Memory allocation failed.\n");
        return NULL;
    }

    int sum = 0;

    for (int i = 0; i < numsSize; i++)
    {
        sum += nums[i];
        answer[i] = sum;
    }

    return answer;
}

File: Typescript/1480.ts
function runningSum(nums: number[]): number[] {
  let runningSum: number[] = [];
  let sum: number = 0;

  for (let i = 0; i < nums.length; i++) {
    sum += nums[i];
    runningSum.push(sum);
  }

  return runningSum;
}

function runningSum2(nums: number[]): number[] {
  const result: number[] = [];

  nums.reduce((prev, next) => {
    result.push(prev + next);
    return prev + next;
  }, 0);

  return result;
}
